File: src/lib/catalog.ts
import type { Mob, Accessory } from "./types";

// The buddy roster. Each buddy's look is drawn deterministically from its `id`,
// so the id is the visual seed — keep it stable (the backend stores it in
// inventory_mobs / profiles.equipped_mob_id). Only the friendly names changed.

export const MOBS: Omit<Mob, "owned">[] = [
  { id: "cat", name: "Pip", price: 0, perk: null },
  { id: "frog", name: "Sprout", price: 120, perk: null },
  { id: "dog", name: "Biscuit", price: 150, perk: null },
  { id: "hollowmark", name: "Pebble", price: 200, perk: null },
  { id: "fox", name: "Sunny", price: 220, perk: "bonusCoins" },
  { id: "penguin", name: "Frosty", price: 260, perk: null },
  { id: "snarejaw", name: "Nibbles", price: 290, perk: null },
  { id: "panda", name: "Bamboo", price: 320, perk: null },
  { id: "nocturne", name: "Twinkle", price: 360, perk: null },
  { id: "bear", name: "Honey", price: 400, perk: "streakShield" },
  { id: "direling", name: "Comet", price: 500, perk: null },
  { id: "elephant", name: "Peanut", price: 600, perk: "bonusCoins" },
  { id: "gloamwyrm", name: "Noodle", price: 660, perk: null },
  { id: "mawborne", name: "Mochi", price: 720, perk: "streakShield" },
  { id: "grimhold", name: "Rocky", price: 820, perk: null },
  { id: "vantamark", name: "Inky", price: 900, perk: "bonusCoins" },
  { id: "embermaw", name: "Blaze", price: 1100, perk: "bonusCoins" },
  { id: "wrathe", name: "Zippy", price: 1500, perk: "streakShield" },
];

export const ACCESSORIES: Omit<Accessory, "owned">[] = [
  { id: "bow", name: "Bow", price: 60 },
  { id: "sunglasses", name: "Cool Shades", price: 70 },
  { id: "headphones", name: "Headphones", price: 80 },
  { id: "party", name: "Party Hat", price: 90 },
  { id: "crown", name: "Crown", price: 200 },
];

export function buddyName(id: string): string {
  return MOBS.find((m) => m.id === id)?.name ?? "Buddy";
}

File: lib/data/catalog.dart
import '../models/models.dart';

/// The canonical catalogue of mobs and accessories. Prices and perks live here
/// in the client; *ownership* and the equipped selection live in Supabase.
/// The `id`s must match what the backend stores (inventory + profile).
class Catalog {
  Catalog._();

  static final List<Mob> mobs = [
    Mob(id: 'cat', name: 'Pixel Cat', emoji: '🐱', price: 0),
    Mob(id: 'dog', name: 'Sunny Pup', emoji: '🐶', price: 150),
    Mob(id: 'fox', name: 'Ember Fox', emoji: '🦊', price: 220, perk: MobPerk.bonusCoins),
    Mob(id: 'bear', name: 'Cozy Bear', emoji: '🐻', price: 400, perk: MobPerk.streakShield),
    Mob(id: 'panda', name: 'Bamboo Panda', emoji: '🐼', price: 320),
    Mob(id: 'elephant', name: 'Grand Elephant', emoji: '🐘', price: 600, perk: MobPerk.bonusCoins),
    Mob(id: 'frog', name: 'Lily Frog', emoji: '🐸', price: 180),
    Mob(id: 'penguin', name: 'Frost Penguin', emoji: '🐧', price: 260),
  ];

  static final List<Accessory> accessories = [
    Accessory(id: 'headphones', name: 'Headphones', emoji: '🎧', price: 80),
    Accessory(id: 'sunglasses', name: 'Sunglasses', emoji: '🕶️', price: 70),
    Accessory(id: 'crown', name: 'Crown', emoji: '👑', price: 200),
    Accessory(id: 'party', name: 'Party Hat', emoji: '🎉', price: 90),
    Accessory(id: 'bow', name: 'Bow Tie', emoji: '🎀', price: 60),
  ];

  static Mob mobById(String id) =>
      mobs.firstWhere((m) => m.id == id, orElse: () => mobs.first);

  static String emojiForMob(String id) => mobById(id).emoji;

  static Accessory? accessoryById(String? id) {
    if (id == null) return null;
    for (final a in accessories) {
      if (a.id == id) return a;
    }
    return null;
  }
}
